### ISSUE TITLE
Released Vassals Ideas

### ISSUE THREAD

**Issue by stackpoint** [TheDivexz/excelmakesbelhappypack#2]
Countries that are released as a vassal (and in general) don't have idea groups in this modpack.

The historical_idea_groups setting controls the ideas that countries spawn with even when AI_USES_HISTORICAL_IDEA_GROUPS is disabled and this modpack voids the historical_idea_groups setting.

**Comment by stackpoint** [comment]
Here are a couple options:

1. Individually write historical_idea_groups for the 830+ tags
- flogi wisely decided that this was a huge waste of time and just voided all the historical_idea_groups instead of trying to fit 16 idea groups for 800+ vanilla tags

2. Setup up generic historical_idea_groups 
- Not actually difficult since you can just use a regular expression or python script to modify all the country files all at once. Would just need to figure out what particular idea groups you would want the AI to pick.

3. Let the spawned country pick the groups in-game and give them the ideas 
- There would be a bunch of ways to do this but it would probably involve the on_actions file: https://github.com/stackpoint/excelmakesbelhappypack/pull/3/commits/57fe8d0cdc2dec0a2c7cb5395b200caf29a8b1c5
Basically there is on_create_vassal and on_country_released that triggers and there you can run events and give the AI something like idea cost reduction for X duration. But idea cost can't be reduced past 90% which means idea costs 25 at minimum while nations spawn with no monarch points.

**Comment by stackpoint** [comment]
I've tested an idea ( https://github.com/stackpoint/excelmakesbelhappypack/pull/3/commits/24553ab3aeec2e362a9055671d57cdfe6c65d0d8 ) where I added a decision that an AI can click for free points when they don't have idea groups filled out but as expected the AI did not always fill the idea groups in a timely fashion even with 90% idea cost reduction and ended up using the monarch points to take tech and other things.

**Comment by mellester** [comment]
My vote would be to use pyhton to genrate 800+ random idea groups for tags

**Comment by stackpoint** [comment]
As an aside, [this](https://github.com/TheDivexz/excelmakesbelhappypack/blob/master/common/countries/Natives.txt#L7) is why the colonial nations don't spawn in with any ideas groups. We don't have to fill out an entire list of groups but adding expansion_ideas will make them spawn in with that idea group at least. Would solve having to set various factors in the idea groups code.

**Comment by mellester** [comment]
As a alternative you can have a event fire that adds the idea group and ideas to the nation just released in question. Using add_idea_group and add_idea  A quick fix would be to directly copy the overlords ideas and start working from there.

**Comment by stackpoint** [comment]
In my testing the AI will pick idea groups within 30 days of being spawned so no need to copy the overlords ideas. After that a decision can be used to grant them those ideas. I have a draft I will upload in a moment.

**Comment by stackpoint** [comment]
See: https://github.com/TheDivexz/excelmakesbelhappypack/pull/5

**Comment by stackpoint** [comment]
See: https://github.com/TheDivexz/excelmakesbelhappypack/issues/11#issuecomment-613474612

**Comment by stackpoint** [comment]
I think there is an expectation that your colonial nation take expansion as their first idea group, whether they can actually afford to colonize or not. Until we figure out the AI logic causing them not to choose it, we should set it manually in historical_idea_groups.

**Comment by stackpoint** [comment]
Released vassals/countries should now get the same number of ideas as the country who released them, after 40-50 days. This is to give the released nation enough time to actually pick their ideas groups.
